test/widgets/subscription_list_test.dart

```diff
@@ -92,8 +92,8 @@ void main() {
   testWidgets('subscription sort', (tester) async {
     await setupStreamListPage(tester, subscriptions: [
       eg.subscription(eg.stream(streamId: 1, name: 'd'), pinToTop: true),
-      eg.subscription(eg.stream(streamId: 2, name: 'c'), pinToTop: false),
-      eg.subscription(eg.stream(streamId: 3, name: 'b'), pinToTop: true),
+      eg.subscription(eg.stream(streamId: 2, name: 'C'), pinToTop: false),
+      eg.subscription(eg.stream(streamId: 3, name: 'B'), pinToTop: true),
       eg.subscription(eg.stream(streamId: 4, name: 'a'), pinToTop: false),
```

Comment:
The main purpose of a test is to make sure we don't accidentally revert your change in the future.

So when writing tests, it's important to check that if you undo your changes, the test would actually fail (and therefore succeed at its purpose). Similarly if you try breaking your changes in a partial way.

Here, there are two `toLowerCase` calls you add. Right now they're on consecutive lines and obviously parallel, and it's unlikely that in that form we'd ever remove one of them without the other — but in the future as things get more complex, they may become separated by many lines of code, and then it becomes entirely likely that'd they'd evolve out of sync with each other. So the test should successfully catch either one of the two `toLowerCase` calls going missing.

As written, I believe this test would catch only one, and not the other.

Reply:
Probably the best way to do this robustly is to add a new test case, rather than edit this one. When one test is testing several different aspects of things, it becomes difficult to be sure what they all are, and therefore difficult to update the test correctly.

For example, it's already hard to tell by looking at this change whether it defeats one of the existing checks this test was intended to make. That's a sign that this test is already checking too many things.

So let's take this in two steps:
* First, one commit splits up this existing test case: one test case to check that the pinned and unpinned sections each get sorted, and one to check that pinned goes before unpinned.
* Then, one commit makes your change to the lib code, and adds a new test case that checks that lower-case vs. upper-case letters are treated the same.